### ISSUE TITLE
fromJSON should support nested directories

### ISSUE THREAD

**Issue by iamogbz** [streamich/memfs#400]
Currently if I want to construct a directory structure like

```js
{
  versions: {
    'v1.1.1': {},
    'v1.2.3': {
      somefile: '.keep',
    },
    'v12.11.10-test-b2': {},
    ...
  }
}
```

I would have to flatten it to

```js
{
  'versions/v1.1.1': {},
  'versions/v1.2.3/somefile': '.keep',
  'versions/v12.11.10-test-b2': {},
  ...
}
```

**Feature Request**

`fromJSON` should be able to build the file system from any valid JSON structure e.g. `mock-fs`

**Comment by pizzafroide** [comment]
@vjpr has [suggested that advanced export/import functionalities could be a separate module](https://github.com/streamich/memfs/issues/48#issuecomment-340404484), so it can be used with real `fs`, as well as with `memfs`.

**Comment by G-Rath** [comment]
I'm going to close this as being fixed by #535, which adds this via `fromNestedJSON`.
